File: Algorithms and Problem Solving Asssignment/isPalindrome.py
def is_palindrome_or_not(palindrome):
    rear_index, front_index = len(palindrome) - 1, 0
    while front_index <= rear_index:
        if palindrome[front_index] != palindrome[rear_index]:
            return 'No it is not a palindrome'
        front_index += 1
        rear_index -= 1
    return 'It is a palindrome'


input_string = input('Enter the string to check whether it is palindrome or not : ')
print(is_palindrome_or_not(input_string))

File: training/java/core/june26_methods/IsPalindrome.java
package june26_methods;
import java.util.Scanner;

public class IsPalindrome {
	
	public static boolean isPalindrome(int num) {
		if (num < 0) 
			num = Math.abs(num);   		int original = num;        
		int rev = 0;

		while (num > 0) {
			int r = num % 10;
			rev = rev * 10 + r;    
			num /= 10;
		}
		return (original == rev);  
	}

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		System.out.println("Enter the number to check whether it's palindrome or not:");
		int num = sc.nextInt();

		System.out.println("The number is : " + (isPalindrome(num) ? "Palindrome" : "Not Palindrome"));
		sc.close();
	}
}
